src/makeCalculator.js

```diff
@@ -4,7 +4,43 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    add(value) {
+      return (this.result += value);
+    },
+
+    subtract(value) {
+      return (this.result -= value);
+    },
+
+    multiply(value) {
+      return (this.result *= value);
+    },
+
+    divide(value) {
+      if (value === 0) {
+        return undefined;
```

Comment:
[CODE KNOWLEDGE] - Returning `undefined` explicitly is not necessary here. If you want to handle division by zero, you should rather throw an error or handle it in a way that aligns with the rest of your application logic.